webdriver/webdriver.go

```diff
@@ -13,28 +15,71 @@ var (
 	startFrameBuffer = flag.Bool("frame_buffer", frameBufferDefault(), "Whether to use a frame buffer")
 	seleniumPath     = flag.String("selenium_path", "", "Path to the selenium standalone binary.")
 	seleniumHost     = flag.String("selenium_host", "localhost", "Host to run selenium on")
-	seleniumPort     = flag.Int("selenium_port", 8888, "Port to run selenium on")
+	seleniumPort     = flag.Int("selenium_port", 0, "Port to run selenium on")
 )
 
 func frameBufferDefault() bool {
 	return runtime.GOOS != "darwin"
 }
 
+func pickUnusedPort() int {
+	addr, err := net.ResolveTCPAddr("tcp", "127.0.0.1:0")
+	if err != nil {
+		panic(err)
+	}
+	l, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		panic(err)
+	}
+	port := l.Addr().(*net.TCPAddr).Port
+	if err := l.Close(); err != nil {
+		panic(err)
+	}
+	return port
+}
+
+// GetWebDriver starts a WebDriver service (server) and creates a remote
+// (client).
+// Make sure to close both the service and the WebDriver instances, e.g.
+//
+// server, driver, err := GetWebDriver()
+// if err != nil {
+//   panic(err)
+// }
+// defer server.Stop()
```

Comment:
Does order matter here? If so, we should mention it. It's easy to forget the order in which defers stack.

Reply:
After some experiment I believe we need to quit the driver session (remote) first, which is also logical (if we kill the whole server process first, we can't really ask the server to close a session). I tweaked the comment.